core/src/main/java/org/stellar/anchor/util/NetUtil.java

```diff
@@ -1,24 +1,43 @@
 package org.stellar.anchor.util;
 
+import static org.stellar.anchor.util.Log.debugF;
 import static org.stellar.anchor.util.StringHelper.isEmpty;
 
 import io.jsonwebtoken.lang.Strings;
 import java.io.IOException;
 import java.net.InetAddress;
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.Objects;
 import okhttp3.Call;
 import okhttp3.Request;
 import okhttp3.Response;
 
 public class NetUtil {
+
   public static String fetch(String url) throws IOException {
-    Request request = OkHttpUtil.buildGetRequest(url);
-    Response response = getCall(request).execute();
 
-    if (response.body() == null) return "";
-    return Objects.requireNonNull(response.body()).string();
+    try {
+      Request request = OkHttpUtil.buildGetRequest(url);
+      Response response = getCall(request).execute();
+      String message =
+          String.format("Error fetching from URL: %s response: %s", url, response.toString());
+
+      // Check if response was unsuccessful (ie not status code 2xx) and throw IOException
+      if (!response.isSuccessful()) {
+        debugF(message);
```

Comment:
Nit: I guess if we're going to throw the exception with the same message, there's no need to log it.